lib/node/http.js

```diff
@@ -165,6 +166,37 @@ class HTTP extends Server {
       });
     });
 
+    // Balance by address
+    this.get('/balance/:address', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const address = valid.str('address');
+
+      enforce(address, 'Address is required.');
+      enforce(!this.chain.options.spv, 'Cannot get balance in SPV mode.');
+
+      const addr = Address.fromString(address, this.network);
+      const coins = await this.node.getCoinsByAddress(addr);
+      const result = {
+        height: this.chain.height,
+        balance: 0,
+        coins: [],
+        lockedBalance: 0,
+        lockedCoins: []
+      };
+
+      for (const coin of coins) {
+        if (coin.covenant.type === rules.types.NONE) {
+          result.balance += coin.value;
+          result.coins.push(coin.getJSON(this.network));
+        } else {
+          result.lockedBalance += coin.value;
+          result.lockedCoins.push(coin.getJSON(this.network));
```

Comment:
See the function `lockBalances()` in `txdb.js` to double-check but I think other covenant types are considered "unlocked", especially `REDEEM` (actually that might be the only other one besides `NONE`)